File: backend/Minimarket.Api/Controllers/AuthController.cs
using Microsoft.AspNetCore.Mvc;
using Minimarket.Api.DTOs;
using Minimarket.Api.Services;

namespace Minimarket.Api.Controllers;

[ApiController]
[Route("api/[controller]")]
public class AuthController(IUserService userService) : ControllerBase
{
    [HttpPost("login")]
    public async Task<ActionResult<LoginResponseDto>> Login([FromBody] LoginRequestDto dto)
    {
        var user = await userService.LoginAsync(dto);
        return user is null ? Unauthorized(new { message = "Credenciales invalidas." }) : Ok(user);
    }
}

File: backend-java/minimarket-api-java/src/main/java/com/minimarket/api/controller/AuthController.java
package com.minimarket.api.controller;

import com.minimarket.api.dto.ApiMessageResponse;
import com.minimarket.api.dto.LoginRequestDto;
import com.minimarket.api.dto.LoginResponseDto;
import com.minimarket.api.service.UserService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/auth")
public class AuthController {

    private final UserService userService;

    public AuthController(UserService userService) {
        this.userService = userService;
    }

    @PostMapping("/login")
    public ResponseEntity<?> login(@RequestBody LoginRequestDto dto) {
        LoginResponseDto user = userService.login(dto);
        if (user == null) {
            return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
                .body(new ApiMessageResponse("Credenciales invalidas."));
        }

        return ResponseEntity.ok(user);
    }
}
